### PR TITLE
fix: Update validations for filter service

### PR DESCRIPTION
**Pull request by pranavrajs**
Fixes https://linear.app/chatwoot/issue/CW-2702/activerecordstatementinvalid-pginvalidtextrepresentation-error-invalid
Fixes https://linear.app/chatwoot/issue/CW-2703/activerecordstatementinvalid-pginvaliddatetimeformat-error-invalid
Fixes https://linear.app/chatwoot/issue/CW-2700/activerecordstatementinvalid-pgsyntaxerror-error-syntax-error-at-or

- refactor filter service for better readability and maintenance 
- Add validations for the following:
  - If an invalid attribute is passed, a custom exception InvalidAttribute will be thrown.
  - If an invalid operator is passed, a custom exception InvalidOperator will be thrown.
  - If an invalid value (currently checking only null check), a custom exception InvalidValue will be thrown.

### CODE REVIEW

**Review by sojan-official — APPROVED** [review]
LGTM

**Comment by pranavrajs** [comment]
@scmmishra Please take a look and review the PR.

File: app/helpers/filter_helper.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      raise CustomExceptions::CustomFilter::InvalidOperator.new(
+        attribute_name: query_hash['attribute_key'],
+        allowed_keys: current_filter['filter_operators']
+      )
+    end
+
+    # Every other filter expects a value to be present
+    if %w[is_present is_not_present].exclude?(query_hash[:filter_operator]) && query_hash['values'].blank?
+      raise CustomExceptions::CustomFilter::InvalidValue.new(attribute_name: query_hash['attribute_key'])
+    end
```
Comment (scmmishra) [inline app/helpers/filter_helper.rb:17]: We could separate this out as a separate method, `validate_query_hash`

```rb
def validate_query_hash(model_filters, query_hash)
  current_filter = model_filters[query_hash['attribute_key']]

  if current_filter.present? && current_filter['filter_operators'].exclude?(query_hash[:filter_operator])
    raise CustomExceptions::CustomFilter::InvalidOperator.new(
      attribute_name: query_hash['attribute_key'],
      allowed_keys: current_filter['filter_operators']
    )
  end

  return unless %w[is_present is_not_present].exclude?(query_hash[:filter_operator]) && query_hash['values'].blank?

  raise CustomExceptions::CustomFilter::InvalidValue.new(attribute_name: query_hash['attribute_key'])
end
```

File: app/helpers/filter_helper.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  # Then we can remove the condition for Contact
+  def handle_additional_attributes(query_hash, filter_operator_value)
+    if filter_config[:entity] == 'Contact'
+      "LOWER(#{filter_config[:table_name]}.additional_attributes ->> '#{query_hash[:attribute_key]}') " \
+        "#{filter_operator_value} #{query_hash[:query_operator]}"
+    else
+      "#{filter_config[:table_name]}.additional_attributes ->> '#{query_hash[:attribute_key]}' " \
+        "#{filter_operator_value} #{query_hash[:query_operator]} "
+    end
+  end
```
Comment (scmmishra) [inline app/helpers/filter_helper.rb:58]: We can abstract the query building logic for the `handle_additonal_attributes` and `default_filter` 

```rb
def handle_additional_attributes(query_hash, filter_operator_value)
  attribute_value = "#{filter_config[:table_name]}.additional_attributes ->> '#{query_hash[:attribute_key]}'"
  build_query_condition(attribute_value, filter_operator_value, query_hash[:query_operator])
end

def default_filter(query_hash, filter_operator_value)
  attribute_value = "#{filter_config[:table_name]}.#{query_hash[:attribute_key]}"
  build_query_condition(attribute_value, filter_operator_value, query_hash[:query_operator])
end

def build_query_condition(attribute_value, filter_operator_value, query_operator)
  if filter_config[:entity] == 'Contact'
    "LOWER(#{attribute_value}) #{filter_operator_value} #{query_operator}"
  else
    "#{attribute_value} #{filter_operator_value} #{query_operator}"
  end
end
```

**Review by scmmishra — COMMENTED** [review]
Overall looks good, added some minor suggestions to improve readability

File: app/helpers/filter_helper.rb (hunk 2 above)
Comment (sojan-official) [reply app/helpers/filter_helper.rb:58]: Have some work done on this in : https://github.com/chatwoot/chatwoot/pull/9048/files 

We will do further refactoring after both these PRs are merged in

**Pull request merged**